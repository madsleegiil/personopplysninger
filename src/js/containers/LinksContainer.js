import React from "react";
import "less/index.less";
import links from "../static/links";
import LinkBox from "../components/LinkBox";

const LinksContainer = () => (
  <React.Fragment>
    {links.map(
      (link, index) =>
        link.visible && (
          <LinkBox
            key={index} // eslint-disable-line react/no-array-index-key
            icon={link.icon}
            header={link.header}
            information={link.information}
            url={link.url}
            linkText={link.linkText}
            kilde={link.kilde}
            infoBoxContent={link.infoBoxContent}
          />
        )
    )}
  </React.Fragment>
);

export default LinksContainer;
