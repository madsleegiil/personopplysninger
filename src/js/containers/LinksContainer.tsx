import React from "react";
import "less/index.less";
import links from "../static/links";
import LinkBox from "../components/LinkBox/LinkBox";

const LinksContainer = () => (
  <React.Fragment>
    {links.map(
      (link: any) =>
        link.visible && (
          <LinkBox
            id={link.id}
            key={link.id}
            icon={link.icon}
            header={link.header}
            information={link.information}
            url={link.url}
            linkText={link.linkText}
            kilde={link.kilde}
            infoBoxContent={link.infoBoxContent}
          />
        )
    )}
  </React.Fragment>
);

export default LinksContainer;
