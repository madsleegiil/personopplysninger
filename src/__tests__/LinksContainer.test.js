import * as React from 'react';
import wrapIntl from 'js/IntlTestHelper';
import LinksContainer from 'js/containers/LinksContainer';
const ReactTestRenderer = require('react-test-renderer');

test('render LinksContainer with content', () => {
  const component = ReactTestRenderer.create(wrapIntl(<LinksContainer />));
  expect(component.toJSON()).toMatchSnapshot();
});
