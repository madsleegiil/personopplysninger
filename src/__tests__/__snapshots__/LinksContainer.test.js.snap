// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render LinksContainer with content 1`] = `
Array [
  <div
    className="BoxContainer box-bottom-margin"
  >
    <div
      className="BoxWithHeader"
    >
      <h1>
        Pensjonsopptjening
      </h1>
      <div
        className="icon-box-wrapper"
      >
        <img
          alt=""
          className="box-icon"
          src="pensjonist.png"
        />
        <div
          className="Box"
        >
          <div
            className="link-box-content"
          >
            <div
              className="link-info"
            >
              NAV har opplysninger om din pensjonsopptjening i en egen tjeneste.
            </div>
            <a
              href="https://tjenester.nav.no/pselv/publisering/dinpensjon.jsf"
              rel="noopener noreferrer"
              target="_blank"
            >
              Gå til Din pensjon
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="BoxContainer box-bottom-margin"
  >
    <div
      className="BoxWithHeader"
    >
      <h1>
        Dine saker
      </h1>
      <div
        className="icon-box-wrapper"
      >
        <img
          alt=""
          className="box-icon"
          src="papirstabel.svg"
        />
        <div
          className="Box"
        >
          <button
            className="information-circle"
            onClick={[Function]}
            type="button"
          >
            <img
              alt="Information"
              src="infomation-circle.svg"
            />
          </button>
          <div
            className="link-box-content"
          >
            <div
              className="link-info"
            >
              Oversikt over dine saker og kommunikasjon med NAV
            </div>
            <a
              href="https://tjenester.nav.no/saksoversikt/app"
              rel="noopener noreferrer"
              target="_blank"
            >
              Gå til saksoversikt
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="BoxContainer box-bottom-margin"
  >
    <div
      className="BoxWithHeader"
    >
      <h1>
        Utbetalinger
      </h1>
      <div
        className="icon-box-wrapper"
      >
        <img
          alt=""
          className="box-icon"
          src="penger.svg"
        />
        <div
          className="Box"
        >
          <div
            className="link-box-content"
          >
            <div
              className="link-info"
            >
              Oversikt over dine utbetalinger fra NAV
            </div>
            <a
              href="https://tjenester.nav.no/utbetalingsoversikt/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Gå til utbetalinger
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="BoxContainer box-bottom-margin"
  >
    <div
      className="BoxWithHeader"
    >
      <h1>
        CV og stillingssøk
      </h1>
      <div
        className="icon-box-wrapper"
      >
        <img
          alt=""
          className="box-icon"
          src="jobbsok.svg"
        />
        <div
          className="Box"
        >
          <div
            className="link-box-content"
          >
            <div
              className="link-info"
            >
              Din CV og lagrede stillingssøk
            </div>
            <a
              href="https://arbeidsplassen.nav.no/minside"
              rel="noopener noreferrer"
              target="_blank"
            >
              Gå til CV/stillingssøk
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
